### PR TITLE
Add "Challenge" and "..." buttons to top of chat windows

### PR DESCRIPTION
**Pull request by jordanhui19**
Placed the username from the header of pm-window in a span with the class "header-username", added a click event for it that brings up the challenge/chat window.
Adds feature requested in issue #1384.

osdchallege

### CODE REVIEW

**Comment by jordanhui19** [comment]
I wasn't able to figure out how to un-highlight the minimize button when the username was hovered. Any tips?

**Comment by Zarel** [comment]
Maybe it should be a separate box below the titlebar, instead of being part of the titlebar.

**Comment by jordanhui19** [comment]
Do you mean something like having a "Challenge" button under the title bar instead of having to click the name to bring up the popup?

**Comment by Zarel** [comment]
Yeah.

**Comment by Zarel** [comment]
(Maybe also have the "More" button next to it.)

**Comment by Zarel** [comment]
I would put it under the header. It also makes sense for it to scroll offscreen when there's a lot of chat, leaving more room for chat.

**Comment by jordanhui19** [comment]
But if you have a lot of chat, wouldn't it be a hassle to have to scroll all the way back up to get the buttons again?

**Comment by jordanhui19** [comment]
Also the chat display is the same size as before, so right now you can still see the same number of messages. The buttons are just in between the header and the pm log, so the overall window is bigger.

**Comment by Zarel** [comment]
> But if you have a lot of chat, wouldn't it be a hassle to have to scroll all the way back up to get the buttons again?

If you have a lot of chat, you can just click on their username. :p

> the overall window is bigger

Yes, this is what I'm trying to avoid.

**Comment by jordanhui19** [comment]
chatscrollhide
chatscrollhide2

So something like this?

**Comment by jordanhui19** [comment]
I updated it so that the buttons to behave the same way as messages do, so scrolling up hides them.
I still think having a hideable section under the header makes more sense since I think losing the buttons while messaging is weird, but either way works.

File: js/client-mainmenu.js [hunk 1]
```diff
@@ -8,8 +8,11 @@
 			'keydown textarea': 'keyDown',
 			'keyup textarea': 'keyUp',
 			'click .username': 'clickUsername',
+			'click .header-username': 'clickUsername',
```
Comment (Zarel) [inline js/client-mainmenu.js]: This part can be reverted, then?

Comment (jordanhui19) [reply js/client-mainmenu.js]: Yeah it can be taken out.
I forgot to remove it.

**Comment by Zarel** [comment]
CI seems to be failing

**Comment by jordanhui19** [comment]
Alright its fixed.

**Comment by Zarel** [comment]
Cool, thanks!

**Pull request merged**